frontend/src/components/widgets/Radio/Radio.tsx

```diff
@@ -14,128 +14,197 @@
  * limitations under the License.
  */
 
-import React from "react"
-import UIRadio from "src/components/shared/Radio"
+import React, { ReactElement, useState, useEffect } from "react"
+import { withTheme } from "@emotion/react"
+import { Radio as UIRadio, RadioGroup, ALIGN } from "baseui/radio"
+import {
+  WidgetLabel,
+  StyledWidgetLabelHelpInline,
+} from "src/components/widgets/BaseWidget"
+import TooltipIcon from "src/components/shared/TooltipIcon"
+import { Placement } from "src/components/shared/Tooltip"
+import { Theme } from "src/theme"
 import { Radio as RadioProto } from "src/autogen/proto"
-import { FormClearHelper } from "src/components/widgets/Form"
-import { WidgetStateManager, Source } from "src/lib/WidgetStateManager"
+import { WidgetStateManager } from "src/lib/WidgetStateManager"
 import { labelVisibilityProtoValueToEnum } from "src/lib/utils"
 
 export interface Props {
   disabled: boolean
+  theme: Theme
   element: RadioProto
   widgetMgr: WidgetStateManager
   width: number
 }
 
-interface State {
-  /**
-   * The value specified by the user via the UI. If the user didn't touch this
-   * widget's UI, the default value is used.
-   */
-  value: number
-}
-
-class Radio extends React.PureComponent<Props, State> {
-  private readonly formClearHelper = new FormClearHelper()
+function Radio(props: Props): ReactElement {
+  const { theme, element, width, widgetMgr } = props
+  let { disabled } = props
+  const { colors, radii } = theme
+  const style = { width }
+  const { horizontal, options, label, labelVisibility, help } = element
 
-  public state: State = {
-    value: this.initialValue,
-  }
-
-  get initialValue(): number {
+  const initialValue = (): number => {
     // If WidgetStateManager knew a value for this widget, initialize to that.
     // Otherwise, use the default value from the widget protobuf.
-    const storedValue = this.props.widgetMgr.getIntValue(this.props.element)
-    return storedValue !== undefined ? storedValue : this.props.element.default
-  }
-
-  public componentDidMount(): void {
-    if (this.props.element.setValue) {
-      this.updateFromProtobuf()
-    } else {
-      this.commitWidgetValue({ fromUi: false })
-    }
+    const storedValue = widgetMgr.getIntValue(element)
+    return storedValue !== undefined ? storedValue : element.default
   }
 
-  public componentDidUpdate(): void {
-    this.maybeUpdateFromProtobuf()
-  }
+  // Tracks value specified by user via UI
+  const [value, setStateValue] = useState(initialValue())
+  // Tracks whether the value was changed by the user
+  const [source, setStateSource] = useState({ fromUi: false })
 
-  public componentWillUnmount(): void {
-    this.formClearHelper.disconnect()
+  const updateFromProtobuf = (): void => {
+    element.setValue = false
```

Comment:
This line makes me uneasy because, while we do this a lot throughout the codebase (as a sort of necessary evil), it feels especially wrong to have a functional component that modifies its input arguments in some way.

I don't have any good suggestions on how to restructure this right now but will think about it and can hopefully come up with something by the time other comments are addressed.